pkgs/by-name/am/amnezia-vpn/package.nix

```diff
@@ -0,0 +1,133 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  kdePackages,
+  qt6,
+  libsecret,
+  xdg-utils,
+  amneziawg-go,
+  openvpn,
+  shadowsocks-rust,
+  cloak-pt,
+  wireguard-tools,
+  callPackage,
+  busybox,
+  sudo,
+}:
+let
+  amnezia-tun2socks = callPackage ./amnezia-tun2socks { };
+  amnezia-xray = callPackage ./amnezia-xray { };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "amnezia-vpn";
+  version = "4.8.2.3";
+
+  src = fetchFromGitHub {
+    owner = "amnezia-vpn";
+    repo = "amnezia-client";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-bCWPyRW2xnnopcwfPHgQrdP85Ct0CDufJRQ1PvCAiDE=";
+    fetchSubmodules = true;
+  };
+
+  postPatch = ''
+    # fix amneziaWG protocol
+    substituteInPlace client/platforms/linux/daemon/wireguardutilslinux.cpp \
+      --replace-fail 'm_tunnel.start(appPath.filePath("../../client/bin/wireguard-go"), wgArgs);' 'm_tunnel.start("${amneziawg-go}/bin/amneziawg-go", wgArgs);'
+
+    # fix service dns flush
+    substituteInPlace service/server/router_linux.cpp \
+      --replace-fail '|| QFileInfo::exists("/usr/lib/systemd/system/nscd.service"))' '|| QFileInfo::exists("/run/systemd/propagate/nscd.service"))'
```

Comment:
This at least is definitely a problem.

Reply:
Yeap, don't know how to fix this another way

Reply:
https://github.com/amnezia-vpn/amnezia-client/blob/dev/service/server/router_linux.cpp#L160-L170 could be rewritten to use dbus or at least `systemctl list-units` to figure out which services actually exist.